modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -657,14 +657,18 @@ void ONNXCompiled::Run(const std::vector<cv::Mat>& ins,
         for (auto &&c_in_pair : params.const_inputs) {
             const auto idx = getIdxByName(in_tensor_info, c_in_pair.first);
             // Puts const input names in sequence for Run
-            // Ort can math input tensors to CNN inputs by names
-            input_names.emplace_back(c_in_pair.first);
+            // Ort can match input tensors to CNN inputs by names
+            // We change data to data from const input
+            // if input_names already contains const input's name
+            const auto it = std::find(input_names.begin(), input_names.end(), c_in_pair.first);
+            if (it == input_names.end()) {
+                input_names.emplace_back(c_in_pair.first);
+            }
             in_tensors.emplace_back(createTensor(this_memory_info,
                                                  in_tensor_info[idx],
                                                  c_in_pair.second.first));
```

Comment:
And what about tensors now? They are not emplaced by constInputs when const_input == input name, right?

Reply:
On the contrary. Const input replaces data with its own if name exist.

Reply:
It is infer2 case. When we call cfgInputNames(with all names) and .ConstInput(with same name).

Reply:
Refactored/fixed. Added test.